src/go/rpk/pkg/cli/cmd/debug/bundle.go

```diff
@@ -0,0 +1,842 @@
+	)
+	command.Flags().StringVar(
+		&logsUntil,
+		"logs-until",
+		"",
+		fmt.Sprintf(`Include log entries on or older than the specified date. %s`, timeHelpText),
+	)
+	command.Flags().StringVar(
+		&logsSizeLimit,
+		"logs-size-limit",
+		"100MiB",
+		"Read the logs until the given size is reached. Multipliers are also supported, e.g. 3MB, 1GiB.",
+	)
+
+	common.AddKafkaFlags(
+		command,
+		&configFile,
+		&user,
+		&password,
+		&mechanism,
+		&enableTLS,
+		&certFile,
+		&keyFile,
+		&CAFile,
+		&brokers,
+	)
+	common.AddAdminAPITLSFlags(command,
+		&adminEnableTLS,
+		&adminCertFile,
+		&adminKeyFile,
+		&adminCAFile,
+	)
+
+	return command
+}
+
+func executeBundle(
+	fs afero.Fs,
+	conf *config.Config,
+	adm sarama.ClusterAdmin,
+	api *admin.AdminAPI,
+	logsSince, logsUntil string,
+	logsLimitBytes int,
+	timeout time.Duration,
+) error {
+	mode := os.FileMode(0755)
+	timestamp := time.Now().Unix()
+	filename := fmt.Sprintf("%d-bundle.zip", timestamp)
+	f, err := fs.OpenFile(
+		filename,
+		os.O_CREATE|os.O_WRONLY,
+		mode,
+	)
+	if err != nil {
+		return fmt.Errorf("couldn't create bundle file: %w", err)
+	}
+	defer f.Close()
+
+	grp := multierror.Group{}
+
+	m := &sync.Mutex{}
+	w := zip.NewWriter(f)
+	defer w.Close()
+
+	ps := &stepParams{
+		fs:      fs,
+		m:       m,
+		w:       w,
+		timeout: timeout,
+	}
+
+	steps := []step{
+		saveKafkaMetadata(ps, adm),
+		saveDataDirStructure(ps, conf),
+		saveConfig(ps, conf),
+		saveCPUInfo(ps),
+		saveInterrupts(ps),
+		saveResourceUsageData(ps, conf),
+		saveNTPDrift(ps),
+		saveSyslog(ps),
+		savePrometheusMetrics(ps, api),
+		saveDNSData(ps),
+		saveDiskUsage(ps, conf),
+		saveLogs(ps, logsSince, logsUntil, logsLimitBytes),
+		saveSocketData(ps),
+		saveTopOutput(ps),
+		saveVmstat(ps),
+		saveIp(ps),
+		saveLspci(ps),
+		saveDmidecode(ps),
+	}
+
+	for _, s := range steps {
+		grp.Go(s)
+	}
+
+	errs := grp.Wait()
+	if errs != nil {
+		err := writeFileToZip(ps, "errors.txt", []byte(errs.Error()))
+		if err != nil {
+			errs = multierror.Append(errs, err)
+		}
+		log.Info(errs.Error())
+	}
+
+	log.Infof("Debug bundle saved to '%s'", filename)
+	return nil
+}
+
+type kafkaMetadata struct {
+	Brokers        []*broker        `json:"brokers"`
+	Topics         []*topic         `json:"topics"`
+	ConsumerGroups []*consumerGroup `json:"consumerGroups"`
+}
+
+type broker struct {
+	Id   int32  `json:"id"`
+	Addr string `json:"addr"`
+}
+
+type topic struct {
+	Name          string             `json:"name"`
+	Config        map[string]*string `json:"config"`
+	NumPartitions int32              `json:"numPartitions"`
+	ReplFactor    int16              `json:"replFactor"`
+}
+
+type consumerGroup struct {
+	Id      string                 `json:"id"`
+	Members []*consumerGroupMember `json:"members"`
+}
+
+type consumerGroupMember struct {
+	Host string `json:"host"`
+	Id   string `json:"id"`
+}
+
+func saveKafkaMetadata(ps *stepParams, admin sarama.ClusterAdmin) step {
```

Comment:
I think this is fine for now but it might be simpler to issue rpk commands directly and just capture the output (rpk cluster offsets, etc), and it'd give us incentive to improve that output.